### DISCUSSION TITLE
[2021.09.06] Queue 구현하기

### DISCUSSION THREAD

**Discussion by pyunji** [pyunji/Hyundai-ITnE-Team_QuadCore#105] [Algorithm]
# 아래에 주어진 조건을 통해 큐를 구현하라
### 필드
- `T data`: 현재 노드의 데이터 저장
- `Node<T> next`: 다음 노드의 주소 저장
### 생성자
  - param: `T data`

### 필드
- `Node<T> first` - 큐의 맨 앞 요소
- `Node<T> last` - 큐의 맨 뒤 요소
### 메서드
- `void add(T item)`: Queue에 item을 추가한다
  - 필드의 갱신 작업 필요
- `T remove()`: Queue의 맨 앞 요소를 반환하고 제거한다.
  - Queue가 비어있을 경우 `NoSuchElementException` 을 발생시킨다
  - 필드의 갱신 작업 필요
- `T peek()`: Queue의 맨 앞 요소를 반환한다.
  - Queue가 비어있을 경우 `NoSuchElementException` 을 발생시킨다
- `boolean isEmpty()`: Queue가 비어있는지 확인한다.

#### ❗ [동영상 참고](https://www.youtube.com/watch?v=W3jNbNGyjMs)

**Comment by pyunji** [comment]
[Queue 구현하기 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/pyunji/pyunji/DataStructure/Basic_Queue.java)

**Comment by hyk339** [comment]
[Queue 구현하기 solved](https://github.com/hyk339/Hyundai-ITnE-Team_QuadCore/blob/master/khy/dataStructure/Basic_Queue.java)

**Comment by chaeyeon4u** [comment]
[Queue 구현하기 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/master/ycy/homework/dataStructure/QueueBasic.java)

**Comment by leechang4** [comment]
[Queue 구현하기 solved](https://github.com/leechang4/Hyundai-ITnE-Team_QuadCore/blob/master/lch/DataStructure/Basuc_Queue.java)
